scripts/clc/item.py

```diff
@@ -0,0 +1,209 @@
+            continue
+    
+        for file in files:
+
+            if (file.startswith(id + '.') or 
+                file.endswith(f'{dom_code}.tif.lyr') or 
+                file.endswith('QGIS.txt',) or 
+                file == f'readme_{id}.txt'):
+
+                asset_files.append(os.path.join(root, file))
+
+    return asset_files
+ 
+def project_bbox(src: rio.io.DatasetReader, dst_crs: rio.CRS = rio.CRS.from_epsg(4326)) -> tuple[float]:
+    bbox = rio.warp.transform_bounds(src.crs, dst_crs, *src.bounds)
+    return(bbox)
+
+def project_data_window_bbox(src: rio.io.DatasetReader, dst_crs: rio.CRS = rio.CRS.from_epsg(4326), dst_resolution: tuple = (0.25, 0.25)) -> tuple[float]:
+     data, transform = rio.warp.reproject(source=src.read(),
+                                          src_transform=src.transform,
+                                          src_crs=src.crs,
+                                          dst_crs=dst_crs,
+                                          dst_nodata=src.nodata,
+                                          dst_resolution=dst_resolution,
+                                          resampling=rio.warp.Resampling.max)
+     
+     data_window = rio.windows.get_data_window(data, nodata=src.nodata)
+     bbox = rio.windows.bounds(data_window, transform=transform)    
+     return bbox
+
+def create_item(img_path: str, data_root: str) -> pystac.Item:
+    clc_name_elements = deconstruct_clc_name(img_path)
+
+    asset_files = get_item_asset_files(data_root, img_path)
+    asset_files = [f for f in asset_files if not f.endswith('aux')]
+    year = clc_name_elements.get('reference_year')
+    props = {'description': ITEM_DESCRIPTION.format(year=year),
+             'created': None,
+             'providers': CLC_PROVIDER.to_dict(),
+    }
+
+    with rio.open(img_path) as img:
+
+        if clc_name_elements['DOM_code']:
+            bbox = project_bbox(img)
+        else:
+            bbox = project_data_window_bbox(img)
+
+        params = {
+            'id': clc_name_elements.get('id'),
+            'bbox': bbox,
+            'geometry': mapping(box(*bbox)),
+            'datetime': None,
+            'start_datetime': datetime(int(year), 1, 1, microsecond=0, tzinfo=UTC),
+            'end_datetime': datetime(int(year), 12, 31, microsecond=0, tzinfo=UTC),
+            'properties': props,
+        }
+
+    item = pystac.Item(**params)
+    
+    for asset_file in asset_files:
+        try:
+            key, asset = create_item_asset(asset_file, DOM_code=clc_name_elements.get('DOM_code'))
+            item.add_asset(
+                key=key,
+                asset=asset,
+            )
+        except KeyError as error:
+            LOGGER.error("An error occured:", error)
+
+    # TODO: "Thumbnail" was originally put at collection level in the template, while it should perhaps be at item level? Individual previews should be added to each item
```

Comment:
The thumbnail was put at collection level because I can only find it in the xml file from sdi at collection level. Unless the data provider provides thumbnail for each item otherwise we can get rid of thumbnail.

Reply:
Agreed. Could be a good opportunity for the data provider to consider revising/recreating the thumbnails, as they are not very consistent (in terms of map styling/visual appearance) on the sdi portal, too